packages/affiliates/test/mocha/dappmining.js

```diff
@@ -0,0 +1,230 @@
+      assert.equal(result.getAttribution("userb", "0x00000000000000000000000000000000000000A1"), "200");
+
+      assert.equal(result.getAttribution("userb", "0x00000000000000000000000000000000000000A2"), "200");
+      assert.equal(result.getAttribution("userc", "0x00000000000000000000000000000000000000A2"), "75");
+
+      assert.equal(result.getAttribution("usera", "0x00000000000000000000000000000000000000A3"), "300");
+      assert.equal(result.getAttribution("userc", "0x00000000000000000000000000000000000000A3"), "75");
+    });
+    it("procesess simple event stream", async function() {
+      const events = [
+        {
+          name: "PositionCreated",
+          args: ["usera", 100, 100],
+          blockNumber: 0
+        }
+      ];
+      const result = await highland(events)
+        .through(
+          utils.ProcessEventStream({
+            startBlock: 0,
+            endBlock: 1
+          })
+        )
+        .toPromise(Promise);
+      assert.equal(result.get("usera"), "100");
+    });
+    it("procesess complex event stream", async function() {
+      const events = [
+        {
+          name: "PositionCreated",
+          args: ["usera", 100, 100],
+          blockNumber: 0
+        },
+        {
+          name: "PositionCreated",
+          args: ["userb", 100, 100],
+          blockNumber: 1
+        },
+        {
+          name: "PositionCreated",
+          args: ["userc", 100, 100],
+          blockNumber: 2
+        }
+      ];
+      const result = await highland(events)
+        .through(
+          utils.ProcessEventStream({
+            startBlock: 0,
+            endBlock: 3
+          })
+        )
+        .toPromise(Promise);
+      assert.equal(result.get("usera"), "300");
+      assert.equal(result.get("userb"), "200");
+      assert.equal(result.get("userc"), "100");
+    });
+    it("gets rewards with simple examples", async function() {
+      const balances = Balances();
+      let attributions = SharedAttributions();
+      const whitelist = ["Dev1", "Dev2"];
+      const totalRewards = "100";
+      balances.add("a", "10");
+      balances.add("b", "20");
+      balances.add("c", "30");
+      balances.add("d", "40");
+      attributions.attribute("a", "Dev1", "5");
+      attributions.attribute("a", "Dev2", "5");
+      attributions.attribute("b", "Dev1", "10");
+      attributions.attribute("b", "Dev2", "10");
+      attributions.attribute("c", "Dev1", "15");
+      attributions.attribute("c", "Dev2", "15");
+      attributions.attribute("d", "Dev1", "20");
+      attributions.attribute("d", "Dev2", "20");
+      let result = await utils.processRewardData({ attributions, balances, whitelist, totalRewards });
+      assert.equal(result["Dev1"], "50");
+      assert.equal(result["Dev2"], "50");
+    });
+  });
+  describe("dataset1", function() {
+    it("should run with dataset", async function() {
```

Comment:
this dataset is based on the  YD-ETH-MAR21 contract, and has some tagged and untagged transactions on mainnet which makes it a decent test.